realm/src/androidTest/java/io/realm/internal/JNIQueryTest.java

```diff
@@ -739,6 +749,110 @@ public void testMinimumDate() {
 
         assertEquals(new Date(0), table.where().minimumDate(0));
 
+        // test for "#1510 Expand date range"
+
+        table.add(new Date(TimeUnit.SECONDS.toMillis(Integer.MAX_VALUE + 1L)));
+        table.add(new Date(TimeUnit.SECONDS.toMillis(Integer.MIN_VALUE - 100L)));
+
+        assertEquals(new Date(TimeUnit.SECONDS.toMillis(Integer.MIN_VALUE - 100L)), table.where().minimumDate(0));
     }
 
+    public void testDateQuery() throws Exception {
+
+        Table table = new Table();
+        table.addColumn(ColumnType.DATE, "date");
+
+        final DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z", Locale.ENGLISH);
+        final Date past = new Date(TimeUnit.SECONDS.toMillis(Integer.MIN_VALUE - 100L));
+        final Date future = new Date(TimeUnit.SECONDS.toMillis(Integer.MAX_VALUE + 1L));
+        final Date distantPast = formatter.parse("0-12-30 00:00:00 +0000");
```

Comment:
I changed my mind.

Tests for distantPast and distantFuture should be replaced with `new Date(Long.MIN_VALUE)` and `new Date(Long.MAX_VALUE)` to avoid combinatorial explosion...